xbmc/addons/kodi-dev-kit/include/kodi/Filesystem.h

```diff
@@ -2029,8 +2055,9 @@ class ATTR_DLL_LOCAL CFile
     // past the number of bytes read, we read (and skipped) a newline sequence.
     char buffer[1025];
     if (CPrivateBase::m_interface->toKodi->kodi_filesystem->read_file_string(
-            CPrivateBase::m_interface->toKodi->kodiBase, m_file, buffer, sizeof(buffer)))
+            CPrivateBase::m_interface->toKodi->kodiBase, m_file, buffer, sizeof(buffer) - 1))
     {
+      buffer[1024] = '\0';
       line = buffer;
```

Comment:
This line is a problem because we don't know how many bytes of the buffer are valid data. For that reason, the buffer should be initialized like `char buffer[1025]{};`.

Reply:
so my strncpy idiom is missing a `memset()`.